src/main/java/org/spongepowered/api/world/WorldBorder.java

```diff
@@ -208,4 +219,93 @@
+     * 
+     * @param border The border whose properties are to be copied
+     */
+    default void copyPropertiesFrom(WorldBorder border) {
+        setCenter(border.getCenter().getX(), border.getCenter().getZ());
+        setDamageAmount(border.getDamageAmount());
+        setDamageThreshold(border.getDamageThreshold());
+        setDiameter(border.getDiameter(), border.getNewDiameter(), border.getTimeRemaining());
+        setWarningDistance(border.getWarningDistance());
+        setWarningTime(border.getWarningTime());
+    }
+
+    interface Builder extends ResettableBuilder<WorldBorder, Builder> {
+
+        /**
+         * Copies the required data from the passed {@code WorldBorder}.
+         * 
+         * @param border The world border whose data is to be copied
+         * @return The builder, for chaining
+         */
+        @Override
+        Builder from(WorldBorder border);
+
+        /**
+         * Sets the diameter of this world border.
+         * 
+         * @param diameter The diameter that this border will have.
+         * @return The builder, for chaining.
+         * @see WorldBorder#setDiameter(double)
+         */
+        Builder diameter(double diameter);
+
+        /**
+         * Sets the centre of this world border.
+         * 
+         * @param x The x-coordinate of the new centre
+         * @param z The z-coordinate of the new centre
+         * @return The builder, for chaining
+         * @see WorldBorder#setCenter(double, double)
+         */
+        Builder center(double x, double z);
+
+        /**
+         * Sets the warning time of this world border.
+         * 
+         * @param time Warning time in seconds
+         * @return The builder, for chaining
+         * @see WorldBorder#setWarningTime(int)
+         */
+        Builder warningTime(int time);
+
+        /**
+         * Sets the warning distance of this world border.
+         * 
+         * @param distance The warning distance in blocks
+         * @return The builder, for chaining
+         * @see WorldBorder#setWarningDistance(int)
+         */
+        Builder warningDistance(int distance);
+
+        /**
+         * Sets the damage threshold of this world border.
+         * 
+         * @param distance The damage threshold in blocks
+         * @return The builder, for chaining
+         * @see WorldBorder#setDamageThreshold(double)
+         */
+        Builder damageThreshold(double distance);
+
+        /**
+         * Sets the damage amount of this world border.
+         * 
+         * @param damage The damage amount
+         * @return The builder, for chaining
+         * @see WorldBorder#setDamageAmount(double)
+         */
+        Builder damageAmount(double damage);
+        
+        /**
+         * Builds the world border from the information given. If the required
+         * information is not given, an {@link IllegalStateException} is thrown.
+         * 
+         * @return The built world border
```

Comment:
Needs a `@throws IllegalStateException reason`, what would HAVE to be set for it to be built?